olm.sh

```diff
@@ -1,44 +1,138 @@
+USAGE:
+  olm.sh <BUILD_VERSION>
+
+EXAMPLE:
+  $ olm.sh 4.0.5
+EOF
+        exit 255
+    fi
+
+    # assign after trim 'v'
+    BUILD_VERSION="${1/v/}"
+    KUBECTL_DIRECTPV="./kubectl-directpv_${BUILD_VERSION}_linux_amd64"
+
+    echo "Downloading required tools"
+
+    OPERATOR_SDK=./operator-sdk
+    if which operator-sdk >/dev/null 2>&1; then
+        OPERATOR_SDK=operator-sdk
+    elif [ ! -f ./operator-sdk ]; then
+        if ! get_operator_sdk; then
+            echo "unable to get operator-sdk; exiting..."
+            exit 1
+        fi
+    fi
+
+    YQ=./yq
+    if which yq >/dev/null 2>&1; then
+        YQ=yq
+    elif [ ! -f ./yq ]; then
+        if ! get_yq; then
+            echo "unable to get yq; exiting..."
+            exit 1
+        fi
+    fi
+
+    if which podman >/dev/null 2>&1; then
+        PODMAN=podman
+    elif which docker >/dev/null 2>&1; then
+        PODMAN=docker
+    else
+        echo "no podman or docker found; please install"
+        exit 255
+    fi
+
+    get_kubectl_directpv
+}
+
+function main() {
+    mkdir -p "resources/base/${BUILD_VERSION}"
+    "${KUBECTL_DIRECTPV}" install -o yaml > "resources/base/${BUILD_VERSION}/directpv.yaml"
+
+    "${OPERATOR_SDK}" generate bundle \
+                      --package "${PACKAGE}" \
+                      --version "${BUILD_VERSION}" \
+                      --deploy-dir "resources/base/${BUILD_VERSION}" \
+                      --manifests \
+                      --metadata \
+                      --output-dir "bundles/redhat-marketplace/${BUILD_VERSION}" \
+                      --channels stable \
+                      --overwrite
+
+    # Annotations to specify OCP versions compatibility.
+    "${YQ}" -i '.annotations."com.redhat.openshift.versions" |= "v4.8-v4.13"' "bundles/redhat-marketplace/${BUILD_VERSION}/metadata/annotations.yaml"
+
+    # Add needed annotations for redhat marketplace
+    "${YQ}" -i '.metadata.annotations."marketplace.openshift.io/remote-workflow" |= "https://marketplace.redhat.com/en-us/operators/minio-directpv-operator-rhmp/pricing?utm_source=openshift_console"' "bundles/redhat-marketplace/${BUILD_VERSION}/manifests/${PACKAGE}.clusterserviceversion.yaml"
+    "${YQ}" -i '.metadata.annotations."marketplace.openshift.io/support-workflow" |= "https://marketplace.redhat.com/en-us/operators/minio-directpv-operator-rhmp/support?utm_source=openshift_console"' "bundles/redhat-marketplace/${BUILD_VERSION}/manifests/${PACKAGE}.clusterserviceversion.yaml"
+
+    # Use SHA Digest for CSI Provisioner Image
+    image="quay.io/minio/csi-provisioner@$(get_image_hash quay.io/minio/csi-provisioner:v3.4.0)"
+    "${YQ}" -i ".spec.install.spec.deployments[0].spec.template.spec.containers[0].image |= (\"${image}\")" "bundles/redhat-marketplace/${BUILD_VERSION}/manifests/${PACKAGE}.clusterserviceversion.yaml"
+
+    # Use SHA Digest for CSI Resizer Image
+    image="quay.io/minio/csi-resizer@$(get_image_hash quay.io/minio/csi-resizer:v1.7.0)"
```

Comment:
can we keep the side car images in a separate global vars, so that we can update it easily when it changes?